File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int left = 0;
        int right = nums.length-1;
        
        while(left<=right){
            int middle = left+ (right-left)/2;
            int potential = nums[middle];
            if(target == potential){
                return middle;
            } else if(target < potential){
                right = middle-1;
            } else left=middle+1;
        }
        
        return left;
        
    }
}

File: 35-search-insert-position/35-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right =len(nums)-1
        
        while left<=right:
            middle= left+ (right-left)//2
            potential = nums[middle]
            if potential == target:
                return middle
            elif potential > target:
                right = middle-1
            else:
                left = middle+1
        
        return left
        